src/modules/flow/http-server/http-server.c

```diff
@@ -789,6 +793,208 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
     return 1;
 }
 
+static void
+rgb_send_packet_cb(struct http_data *mdata, struct sol_flow_node *node)
+{
+    sol_flow_send_rgb_packet(node,
+        SOL_FLOW_NODE_TYPE_HTTP_SERVER_RGB__OUT__OUT, &mdata->value.rgb);
+}
+
+static int
+rgb_response_cb(struct http_data *mdata, struct sol_buffer *content, bool json)
+{
+    if (json) {
+        return sol_buffer_append_printf(content,
+            "{\"red\":%" PRIu32 ",\"green\":%" PRIu32 ",\"blue\":%" PRIu32
+            ",\"red_max\":%" PRIu32 ",\"green_max\":%" PRIu32
+            ",\"blue_max\":%" PRIu32 "}",
+            mdata->value.rgb.red, mdata->value.rgb.green, mdata->value.rgb.blue,
+            mdata->value.rgb.red_max, mdata->value.rgb.green_max,
+            mdata->value.rgb.blue_max);
+    }
+
+    return sol_buffer_append_printf(content, "#%X%X%X", mdata->value.rgb.red,
```

Comment:
%02x, otherwise if number is <= 0xf it will take a single digit. Like 0 will produce "#000" instead of "#000000".